packages/core/src/lib/modal.svelte

```diff
@@ -60,19 +70,21 @@ onMount(() => {
 });
 </script>
 
+<svelte:window on:keydown={handleEscapePress} />
+
 <div
   class={cx(
     'fixed left-0 top-0 z-50 flex h-full w-full items-center justify-center bg-black bg-opacity-40',
     {
       invisible: !open,
     }
   )}
-  role="button"
-  tabindex="0"
-  on:click={handleBackgroundClick}
-  on:keydown={handleBackgroundKey}
+  role="dialog"
+  tabindex="-1"
```

Comment:
I'm not 100% sure this `tabindex` is necessary; a `div` shouldn't be in the document's tab list to begin with. Will double check MDN on this, though, since maybe it has an effect on children?